src/coreclr/tools/superpmi/superpmi/superpmi.cpp

```diff
@@ -380,334 +372,367 @@ int __cdecl main(int argc, char* argv[])
-                            // Otherwise add it in the failingMCList here.
-                            if (o.details == nullptr)
-                            {
-                                failingToReplayMCL.AddMethodToMCL(reader->GetMethodContextIndex());
-                            }
-
-                            break;
-                        case NearDifferResult::SuccessWithoutDiff:
-                            break;
-                        case NearDifferResult::Failure:
-                            if (o.mclFilename != nullptr)
-                                failingToReplayMCL.AddMethodToMCL(reader->GetMethodContextIndex());
-
-                            break;
-                    }
+                    failingToReplayMCL.AddMethodToMCL(reader->GetMethodContextIndex());
                 }
 
-                if (o.details != nullptr)
+                if ((res.Result == ReplayResult::Miss) || (res2.Result == ReplayResult::Miss))
                 {
-                    PrintDiffsCsvRow(
-                        detailsCsv,
-                        reader->GetMethodContextIndex(),
-                        mcb.size,
-                        res, res2,
-                        /* hasDiff */ diffResult != NearDifferResult::SuccessWithoutDiff);
+                    missingCount++;
                 }
-            }
-            else
-            {
+
                 if (o.details != nullptr)
                 {
-                    PrintReplayCsvRow(
-                        detailsCsv,
-                        reader->GetMethodContextIndex(),
-                        mcb.size,
-                        res);
+                    if (o.applyDiff)
+                    {
+                        PrintDiffsCsvRow(
+                            detailsCsv,
+                            reader->GetMethodContextIndex(), mcb.size,
+                            res, res2,
+                            /* hasDiff */ false);
+                    }
+                    else
+                    {
+                        PrintReplayCsvRow(detailsCsv, reader->GetMethodContextIndex(), mcb.size, res);
+                    }
                 }
             }
-        }
-        else
-        {
-            failToReplayCount++;
-
-            // Methods that don't compile due to missing JIT-EE information
-            // should still be added to the failing MC list but we don't create MC repro for them.
-            if (o.mclFilename != nullptr)
-            {
-                failingToReplayMCL.AddMethodToMCL(reader->GetMethodContextIndex());
-            }
 
-            if ((res.Result == ReplayResult::Miss) || (res2.Result == ReplayResult::Miss))
-            {
-                missingCount++;
-            }
-
-            if (o.details != nullptr)
+            // Write a progress message if necessary.
+            const int progressInterval = 500;
+            if ((jittedCount > 0) && (jittedCount % progressInterval == 0))
```

Comment:
The progress output was moved to the end of the loop because it made all the progress output scenarios work better.